### DISCUSSION TITLE
AG Crashes Logic (M1 MacBook/OSX 14.4.1/Logic 10.8.1)

### DISCUSSION THREAD

**Discussion by mbruhouse** [apohl79/audiogridder#1364] [Help]
Since updating to Sonoma 14.4.1 and Logic 10.8.1 loading Audiogridder on an audio channel crashes logic with the error window "An Audio Unit plug-in reported a problem which might cause the system to become unstable. Please quit and restart Logic". Error is consistent.

AG launches fine under Logic opened using Rosetta, and natively with Reaper and Luna. I've searched high and low for answers specific to this issue and have come up empty handed, so I'm curious if it's an issue on my end or if maybe I've not been searching in the right place. Any help is appreciated.

So far I've tried trashing Logic preferences, deleting caches, and a few other fixes for similar Juce (not AG specific) errors, including unchecking Midi 2 in Logic's preferences. 

Thanks!

<img width="342" alt="Screenshot 2024-04-28 at 6 14 09 AM" src="https://github.com/apohl79/audiogridder/assets/168340129/f173a124-0ae5-408a-a573-3b1e12d9e7b6">

**Comment by mbruhouse** [comment]
Some updated info in the off chance anyone reads this or has any tips.

For reasons I've been unable to determine, AG started working natively again without crashing for awhile (it was after I'd installed a different plugin, Soundtoys Micro-shift if that has any bearing on things), and things were ok for a bit. Now it is crashing again but only when I've loaded AG on a stereo track, saved and closed the session, then re-opened it later. 

I've been able to consistently reproduce this issue. Loaded on to mono tracks, everything works fine every time. Loaded on to a stereo track, things are fine while the session is running, but upon reopening, I get the plugin crash modal. Clicking recover saves the session from closing, but the AG instance on said stereo track opens with no GUI. All instances of AG on mono tracks remain working as normal.

EDIT: Upon further exploration, it seems as though the crash happens if AG is placed on the first track of the session. The crash will happen, and you can recover and every other instance of AG will work as normal but the first instance will open with no GUI. Very strange!

**Reply by ABiguet** [reply]
Hi, same issue here. Did you find any solutions to solve this problem ? 
My setup :
- Serveur : Windows 11 with VST plugins
- Client : MacOS Sonoma, M2, Logic Pro

**Comment by osxdude** [comment]
Just had this with a fresh AudioGridder install. I did install the server on the same computer, but that shouldn't matter, no? (I'd like to use server on another machine) I get the crash modal no matter whether I select mono or stereo in a new file in the latest Logic Pro (11.0.1) on Sonoma 14.6.1.

**Comment by mbruhouse** [comment]
Hey, sorry to say I have yet to fully understand how or why this happens. I got rid of a bunch of older non-Silicon plugins and that helped for awhile but I still get this randomly and I think it's just on a plugin by plugin basis. Gold 5 from Acustica started doing it after the last update, and rolling it back fixed it. I've been having it with Softube VCA compressor as well.

Every time I think I've got a handle on a consistent way to reproduce it, something new happens. But I will say since getting rid of anything that wasn't fully silicon compatible made it way less frequent and I've been able to stop having to open in Rosetta as well.

**Comment by skonghi** [comment]
I have a similar problem. I can't use AG as the first plugin in the first audio track of a project, otherwise it crashes logic and asks to recover AG, which turns it into a blank screen. The way I made it work is to insert a plugin before AG, but not every plugin works. I recently updated my MacOS to Ventura 13.6.9, that's the most probable cause. Using LP 11.0.1

**Comment by ABiguet** [comment]
The solution I've found :

1. Open an empty project in Logic Pro.
2. Create an instrument track that is not AG (for example, an Arturia instrument works well in my case).
3. Add the AG track afterwards, making sure it is launched properly.
4. Close the project in the File menu.
5. Ensure the AudioGridderPluginTray icon is visible.
6. At this point, you can create or open any project that contains the AG plugin.

Hope this helps!

**Reply by Blandystudio** [reply]
that works for me , strange but i'll take it ,thanks ! ;)

update works ..for a bit then same problem

**Reply by pmdevita** [reply]
I wonder if it's some kind of timing issue, if I load a small synth plugin as the first plugin, it crashes, but if I load a larger sample library plugin, then it boots up

**Comment by Blandystudio** [comment]
I's DEFO the AU componant causing problems for me Apple M1 Max Sonoma latest logic x , My Ableton Live latest 12 is rocking the vst audiogridder plug but won't even show the AU ,

thoughts ?

**Comment by mbruhouse** [comment]
This continues to be a somewhat baffling, intermittent annoyance. I've recently upgraded to a new MacBook Pro (M4 Pro, 48gb RAM, Sequoia 15.3.1, Logic Pro 11.1.2), and after getting it all set up, I set to work on getting my audio stuff up and running and installing plugins. On a fresh install the crashing when adding the Audiogridder plugin started happening again. GAH.

I was careful to only install AU plugins wherever possible, and took any and all non-M-series compatible plugins out of the install process, so it isn't that. The trick a couple posts up didn't work either. Loading the plugin on an instrument channel DID bring up the plugin without a crash but didn't fix the issue elsewhere....which was also odd. 

The only thing I could think of that could maybe be the issue was that the server Mac mini was essentially running headless and I was just using the screen sharing application to remote in when I needed to do anything on that machine, so I grabbed an HDMI cable, plugged it into the mini and now the plugin didn't crash Logic, but it loaded without a GUI, just a blank window. Very odd.

I've since added an HDMI dummy connector to the mini and either that fixed it or was just a total coincidence and happened just as the plugin decided to work as normal, but the problem has since gone away.

Take whatever you want from all of that, I have no way of 100% knowing this was the fix or not, as last time it just randomly stopped happening and I could never figure out exactly what I did to fix it.

**Comment by mbruhouse** [comment]
Some more discoveries and findings. Not sure if this had anything to do with some of my previous crashes, but I've noticed in Logic that a session using AG will crash upon loading if I don't have my Audio input and output explicitly set, like if I've unplugged my laptop from my interface to take along to work remotely, and boot up Logic and get "previous device cannot be found" modal. You can tell by looking in your audio settings in Logic, if the output and input devices have parenthesis around them to indicate that these aren't what you initially set, but what Logic defaulted to when it couldn't find the intended device. 

Every time results in a crash. Resetting them manually (they won't be in parenthesis in the settings dialog) will fix it.

Kind of an odd bug but there you go.

**Comment by apohl79** [comment]
Same answer as on the other thread. - I think the problem is the crash reporting framework being used (sentry). Logic is sandboxing plugins now and the sandbox does not like sentry. Try turning crash reporting off in AG. Maybe that helps.

**Reply by pmdevita** [reply]
Seems to help for me, usually with enough attempts on some different projects I can eventually get it to open, from there you can disable it here
<img width="681" height="519" alt="image" src="https://github.com/user-attachments/assets/f7ca9e6e-bbfd-4d36-8c28-cd41cae20c00" />
Haven't had trouble getting it to start since so that seems to be the issue, thank you!
